### PR TITLE
OJ-2214: Store HMRC response into database

### PR DESCRIPTION
**Pull request by surajkumar**
## Proposed changes

* Updated state machine to save an attempt when they are either successful or if the match fails. The state machine will no longer count any failure (i.e. internal errors) as an attempt.
* Removed unable to save user state as errors are already captured by default. 
* Updated nino-attempts table to include status and text
* Updated attempt checking to be based on number of records
* Updated match-handler.ts to return the status code as a String to allow DynamoDB to save this
* Updated HMRC error block to capture errors correctly in state machine

* Changed tests to use different mock data between them. 

### Issue tracking
- [OJ-2214](https://govukverify.atlassian.net/browse/OJ-2214)

[OJ-2214]: https://govukverify.atlassian.net/browse/OJ-2214?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/matching/src/matching-handler.ts [hunk 1]
```diff
@@ -26,12 +26,12 @@ export class MatchingHandler implements LambdaInterface {
       });
       try {
         return {
-          status: response.status,
+          status: response.status.toString(),
```
Comment (jkunle) [inline lambdas/matching/src/matching-handler.ts:29]: Is this change really necessary?

Comment (surajkumar) [reply lambdas/matching/src/matching-handler.ts:29]: Yes, you cannot pass anything that isn't a string to DynamoDB so it won't save. If you want to save a number e.g. 200, it has to be passed as "200"

**Review by jkunle — APPROVED** [review]
